### PR TITLE
carli/2267_remove_distance_measurement_from_pv_preview

### LINKED ISSUE #2267: should not have distance measurement labels on interactive pv image
**Describe the bug**
reported by @jott3077 

When there is an interactive pv preview widget, if we enable distance measurement on the main image (left image in the screenshot), we will see "0.000000" label rendered on the interactive pv image (right image in the screenshot). On the static pv image generated via the pv generator, it works fine (middle image in the screenshot).

Screenshot 2023-10-10 at 09 16 03

**To Reproduce**
1. load an image cube and enable interactive pv preview
2. enable distance measurement tool in the main image viewer
3. make a measurement
4. we will see "0.000000" label appears on the interactive pv image.

**Expected behavior**
no distance measurement labels on the interactive pv image

**Platform info (please complete the following information):**
 - OS [e.g. macOS Monterey]: macOS Ventura
 - Browser [e.g. chrome, safari, electron app]: electron app
 - Browser version [e.g. 22]: 
 - Backend branch [e.g. dev, v3b2 release]: v4-stable, dev
 - Frontend branch [e.g. dev, v3b2 release]: v4-stable, dev

### PR DESCRIPTION
**Pull request by crocka**
**Description**

Closes #2267. Removed distanceMeasuring store object is the frame is a preview frame.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] e2e test passing / corresponding fix added / new e2e test created
- [x] changelog updated / no changelog update needed
- [x] ~~protobuf updated to the latest dev commit~~ / no protobuf update needed
- [x] ~~protobuf version bumped~~ / protobuf version not bumped
- [x] `BackendService` unchanged / ~~`BackendService` changed and corresponding ICD test fix added~~

### CODE REVIEW

File: src/components/ImageView/RegionView/RegionViewComponent.tsx [hunk 1]
```diff
@@ -739,7 +739,7 @@ export class RegionViewComponent extends React.Component<RegionViewComponentProp
         }
 
         let cursor: string;
-        if (regionSet.mode === RegionMode.CREATING || AppStore.Instance?.activeLayer === ImageViewLayer.DistanceMeasuring) {
+        if ((regionSet.mode === RegionMode.CREATING || AppStore.Instance?.activeLayer === ImageViewLayer.DistanceMeasuring) && !frame.isPreview) {
```
Comment (YuHsuan-Hwang) [inline src/components/ImageView/RegionView/RegionViewComponent.tsx]: Currently we can draw regions with the menu. I think we either disable drawing regions or keep the crosshair cursor style for the `CREATING` region mode.

File: src/stores/Frame/FrameStore.ts [hunk 2]
```diff
@@ -2964,6 +2966,7 @@ export class FrameStore {
 
     @action setIsPreview = (isPreview: boolean) => {
         this.isPreview = isPreview;
+        this.distanceMeasuring = null;
```
Comment (YuHsuan-Hwang) [inline src/stores/Frame/FrameStore.ts]: Do we need to check if `isPreview` is true? If `setIsPreview(false)` is called, do we enable distance tool?

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]

File: src/stores/Frame/FrameStore.ts [hunk 3]
```diff
@@ -1222,9 +1223,9 @@ export class FrameStore {
 
         this.stokesFiles = [];
 
-        this.isPreview = false;
+        this.isPreview = frameInfo.preview;
 
-        this.distanceMeasuring = new DistanceMeasuringStore();
+        this.distanceMeasuring = !frameInfo.preview && new DistanceMeasuringStore();
```
Comment (YuHsuan-Hwang) [inline src/stores/Frame/FrameStore.ts]: Please initialize it to null instead of false.

File: src/stores/Frame/FrameStore.ts [hunk 4]
```diff
@@ -1222,9 +1223,9 @@ export class FrameStore {
 
         this.stokesFiles = [];
 
-        this.isPreview = false;
+        this.isPreview = frameInfo.preview;
```
Comment (YuHsuan-Hwang) [inline src/stores/Frame/FrameStore.ts]: If preview is included in `this.frameInfo`, how about we change `this.isPreview` to a `@computed getter`?

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]

**Review by YuHsuan-Hwang — APPROVED** [review]

**Review by loveluthien — APPROVED** [review]
It works well.

**Comment by YuHsuan-Hwang** [comment]
Require final check for e2e tests.

**Review by kswang1029 — APPROVED** [review]
looks good. no regression from e2e tests.

**Pull request merged**